Adds manage tokens guide (#160)
moondance-labs/tanssi-docs: builders/manage/dapp/manage-tokens.md

---
title: Manage Tokens
description: Learn how the appchain governor can manage the token of a Tanssi-powered appchain using the dApp to mint them, configure the EIP-1559 fee market, and more.
---

# Manage Tokens

## Introduction {: #introduction }

> Any appchain deployed through Tanssi is sovereign and free to define the governance model that best fits its use case. The appchain governor has superpowers over the chain administration, meaning privileged functions, such as updating the runtime, fall within his reach.

Reviewer: ```suggestion
Any appchain deployed through Tanssi is sovereign and free to define the governance model that best fits its use case. The appchain governor has superpowers over the chain administration, meaning privileged functions, such as updating the runtime, fall within their reach.
```

🙂

There are some actions related to the native token management that are available to the appchain governor on the Tanssi dApp:

- **Mint Tokens** - mints new tokens, increasing the total supply
- **Update Balances** - increases or decreases the balance of an account, affecting the total supply
- **Transfer Tokens** - executes a forced token transfer from one account to another

> - **Configure Gas Dynamics** - only available on EVM-compatible appchains, this action changes the [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md){target=\_blank} configuration, affecting the transaction pricing mechanism

Reviewer: I just realized that sometimes we do "Mint Tokens" (both caps) other times we do "Mint tokens" (first word caps) - so we need to start making sure we follow a standard here. Let's just do first word caps

In this guide, you'll learn how to execute the previously listed actions using the [Tanssi dApp](https://apps.tanssi.network){target=\_blank}.

## Checking Prerequisites {: #checking-prerequisites }

For the examples in this guide, you will need to have the following:

- A Tanssi appchain (Snap or Dedicated)
- The account you used when registering the appchain, imported in any of the [supported wallets](/builders/deploy/dapp/#supported-wallets){target=\_blank}

> - Your appchain's Sudo account, also imported in any of the [supported wallets](/builders/deploy/dapp/#supported-wallets){target=\_blank}. 

Reviewer: ```suggestion
- Your appchain's Sudo account, also imported in any of the [supported wallets](/builders/deploy/dapp/#supported-wallets){target=\_blank}
```

!!! note
    The appchain's registration account is always a Substrate one, whereas the appchain's Sudo account depends on the chain type. If the chain is EVM-compatible, then the Sudo account will be an Ethereum type and, otherwise, a Substrate type.

### Retrieving the Registration Account {: #retrieving-registration-account }

If you're unsure what your registration account is, you can query it directly from the Tanssi orchestrator chain, which keeps records of every registered appchain. To do so, head to the **Chain state** section on the Polkadot.js Apps connected to the orchestrator chain for [snap appchains](https://polkadot.js.org/apps/?rpc=wss://fraa-flashbox-2607-rpc.a.stagenet.tanssi.network#/chainstate){target=\_blank} or [dedicated appchains](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Ffraa-dancebox-rpc.a.dancebox.tanssi.network#/chainstate){target=\_blank}, and take the following steps:

1. Select the **registrar** storage module
2. Select **registrarDeposit**

> 3. Insert your appchain id

Reviewer: ```suggestion
3. Insert your appchain ID
```

4. Press **+** icon

You'll see the registration account at the bottom.

> ![Locating your registration account](/images/builders/manage/dapp/token-management/token-management-1.webp)

Reviewer: I'm fine with this being here for now due to the sake of time. But I wonder if we should extract this to it's own short guide and just link back to it as needed. What do you think?

Author: Agreed. The same applies to "retrieving the sudo account".
I'll create a ticket to make these sections a snippet at least

### Retrieving the Sudo Account {: #retrieving-sudo-account }

If you're unsure what your Tanssi appchain's Sudo account is, you can find it in your [Tanssi Dashboard](https://apps.tanssi.network){target=\_blank} underneath the **Properties** section.

![Locating your Sudo address on apps.tanssi.network](/images/builders/manage/dapp/token-management/token-management-2.webp)

!!! warning
    You should always protect your Sudo account key with the utmost security precautions, as it grants privileged access to your Tanssi appchain.

## Accessing the Token Management Panel {: #accesing-token-management-panel }

The Tanssi dApp implements a smooth interface, allowing the appchain governor to access and execute privileged functions. To do so, head to the [Tanssi dApp](https://apps.tanssi.network/){target=\_blank}, click on the **Manage** button, and then, again click on the **Token Management** button.

![Accessing the token management panel](/images/builders/manage/dapp/token-management/token-management-3.webp)

Now you have direct access to the actions presented in the introduction:

1. **Mint Tokens**
2. **Update Balances**
3. **Transfer Tokens**

> 4. **Configure Gas Dynamics** 

Reviewer: ```suggestion
4. Configure **Gas Dynamics** 
```
Configure shouldn't be bold as it's not in the UI

> 4. **Configure Gas Dynamics** 

Reviewer: We should really only use numbered lists when there are steps to take in a particular order, if there is no need for the numbers we should use bulleted lists. I think a single arrow pointing to the panel would be sufficient. 

This can be addressed later on

Author: (Adding this as a subtask in the same ticket to change the screenshots)

![The token management panel](/images/builders/manage/dapp/token-management/token-management-4.webp)

!!! note

>     If you don't see a card with your appchain's details, make sure to comply with the [prerequisites](#checking-prerequisites).

Reviewer: Do you mean the **Properties** section?

Author: No, I meant the prerequisites section.
I know that complying with the prerequisites is a requisite, but the appchain will be visible in the dashboard only if everything is well configured. Otherwise, double-check the requirements.

Reviewer: yes, but when you refer to "card" are you referring to the "Properties" card on the dashboard or this section of the dashboard? If so, can you please update the text to tell them what they're looking at

Author: Hmm, I see it is ambiguous. Rephrased!

### Minting Tokens {: #minting-tokens }

In the **Token Management** panel, click on the **Mint Tokens** button, and then:

1. Insert the address that will hold the newly minted tokens
2. Insert the amount of tokens to mint
3. Click on **Mint**

![Mint Tokens](/images/builders/manage/dapp/token-management/token-management-5.webp)

You'll be asked to sign the transaction with the appchain's governor account. Once the transaction has gone through, the destination account's balance will have been increased by the desired amount.

!!! note

>     The destination address must be Ethereum type if the chain is EVM-compatible and a Substrate type otherwise.

Reviewer: I'd move this note up to be nested under step 1

Reviewer: same applies to other places

### Updating Balances {: #updating-balances }

> 

Reviewer: This is another way to mint tokens? Or is this simply sending tokens from the existing governor balance?

Author: It's just another way to mint or burn tokens (you can also decrease a balance)

---
title: Manage Tokens
description: Learn how the appchain governor can manage the token of a Tanssi-powered appchain using the dApp to mint them, configure the EIP-1559 fee market, and more.
---

# Manage Tokens

## Introduction {: #introduction }

> Any appchain deployed through Tanssi is sovereign and free to define the governance model that best fits its use case. The appchain governor has superpowers over the chain administration, meaning privileged functions, such as updating the runtime, fall within their reach.

Reviewer 2: ```suggestion
Any appchain deployed through Tanssi is sovereign and free to define the governance model that best fits its use case. The appchain governor has superpowers over the chain administration. Consequently, they can call privileged functions, such as updating the runtime and managing native token-related operations, among other actions.
```

There are some actions related to the native token management that are available to the appchain governor on the Tanssi dApp:

- **Mint tokens** - mints new tokens, increasing the total supply
- **Update Balances** - increases or decreases the balance of an account, affecting the total supply
- **Transfer Tokens** - executes a forced token transfer from one account to another

> - **Configure Gas Dynamics** - only available on EVM-compatible appchains, this action changes the [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md){target=\_blank} configuration, affecting the transaction pricing mechanism

Reviewer: can you fix the capitalization on these too please!

In this guide, you'll learn how to execute the previously listed actions using the [Tanssi dApp](https://apps.tanssi.network){target=\_blank}.

## Checking Prerequisites {: #checking-prerequisites }

For the examples in this guide, you will need to have the following:

> - A Tanssi appchain (Snap or Dedicated)

Reviewer 2: As this becomes repetitive, we should move this to a text snippet. @eshaben wdyt?

Author: Yes! Already created a ticket to address this:
https://opslayer.atlassian.net/jira/software/c/projects/DT/boards/108?selectedIssue=DT-213

- The account you used when registering the appchain, imported in any of the [supported wallets](/builders/deploy/dapp/#supported-wallets){target=\_blank}
- Your appchain's Sudo account, also imported in any of the [supported wallets](/builders/deploy/dapp/#supported-wallets){target=\_blank}

!!! note
    The appchain's registration account is always a Substrate one, whereas the appchain's Sudo account depends on the chain type. If the chain is EVM-compatible, then the Sudo account will be an Ethereum type and, otherwise, a Substrate type.

> ### Retrieving the Registration Account {: #retrieving-registration-account }

Reviewer 2: Similar here

If you're unsure what your registration account is, you can query it directly from the Tanssi orchestrator chain, which keeps records of every registered appchain. To do so, head to the **Chain state** section on the Polkadot.js Apps connected to the orchestrator chain for [snap appchains](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Ffraa-flashbox-rpc.a.stagenet.tanssi.network#/chainstate){target=\_blank} or [dedicated appchains](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Ffraa-dancebox-rpc.a.dancebox.tanssi.network#/chainstate){target=\_blank}, and take the following steps:

1. Select the **registrar** storage module
2. Select **registrarDeposit**
3. Insert your appchain ID
4. Press **+** icon

You'll see the registration account at the bottom.

> ![Locating your registration account](/images/builders/manage/dapp/token-management/token-management-1.webp)

Reviewer 2: Remove LFD banner so that the image is longer lived

### Retrieving the Sudo Account {: #retrieving-sudo-account }

If you're unsure what your Tanssi appchain's Sudo account is, you can find it in your [Tanssi Dashboard](https://apps.tanssi.network){target=\_blank} underneath the **Properties** section.

> ![Locating your Sudo address on apps.tanssi.network](/images/builders/manage/dapp/token-management/token-management-2.webp)

Reviewer 2: Remove LFD banner so the image is longer lived

Author: The same here, already created a ticket to update the screenshots and use the demo chain instead

!!! warning
    You should always protect your Sudo account key with the utmost security precautions, as it grants privileged access to your Tanssi appchain.

## Accessing the Token Management Panel {: #accesing-token-management-panel }

The Tanssi dApp implements a smooth interface, allowing the appchain governor to access and execute privileged functions. To do so, head to the [Tanssi dApp](https://apps.tanssi.network/){target=\_blank}, click on the **Manage** button, and then, again click on the **Token Management** button.

> ![Accessing the token management panel](/images/builders/manage/dapp/token-management/token-management-3.webp)

Reviewer 2: If the image has numbered arrows we should have a numbered list

Now you have direct access to the actions presented in the introduction:

> 1. **Mint tokens**

Reviewer: since this is referring to a UI element, you can keep capitalization consistent with what users will see on the dashboard so this can stay as "Mint Tokens"

2. **Update Balances**
3. **Transfer Tokens**
4. Configure **Gas Dynamics** 

![The token management panel](/images/builders/manage/dapp/token-management/token-management-4.webp)

!!! note
    If you don't see a card with your appchain's details, make sure to comply with the [prerequisites](#checking-prerequisites).

### Minting Tokens {: #minting-tokens }

> The appchain governor can mint new tokens, increasing its total supply. In the **Token Management** panel, click on the **Mint tokens** button, and then:

Reviewer: ```suggestion
The appchain governor can mint new tokens, increasing its total supply. To do so, in the **Token Management** panel, click on the **Mint tokens** button, and then:
```

1. Insert the address that will hold the newly minted tokens

    !!! note
        The destination address must be Ethereum type if the chain is EVM-compatible and a Substrate type otherwise.

2. Insert the amount of tokens to mint
3. Click on **Mint**

![Mint tokens](/images/builders/manage/dapp/token-management/token-management-5.webp)

You'll be asked to sign the transaction with the appchain's governor account. Once the transaction has gone through, the destination account's balance will have been increased by the desired amount.

### Updating Balances {: #updating-balances }

> The appchain governor can increase or decrease the balance of any account, thus affecting the total supply. In the **Token Management** panel, click on the **Update Balances** button, and then:

Reviewer: ```suggestion
The appchain governor can increase or decrease the balance of any account, thus affecting the total supply. To do so, in the **Token Management** panel, click on the **Update Balances** button, and then:
```

1. Insert the address that will hold the newly minted tokens. Once you enter the address, its current balance will be displayed

    !!! note
        The destination address must be Ethereum type if the chain is EVM-compatible and a Substrate type otherwise.
    
2. Insert the new balance the address will hold
3. Click on **Update**

![Update Balances](/images/builders/manage/dapp/token-management/token-management-6.webp)

You'll be asked to sign the transaction with the appchain's governor account. Once the transaction has gone through, the destination account's balance will reflect exactly the desired amount, regardless of previous holdings.

### Executing Forced Transfers {: #forced-transfers }

> The appchain governor can reassign balances, forcing a token transfer from one account to another. In the **Token Management** panel, click on the **Transfer Tokens** button, and then:

Reviewer: ```suggestion
The appchain governor can reassign balances, forcing a token transfer from one account to another. To do so, in the **Token Management** panel, click on the **Transfer Tokens** button, and then:
```

1. Insert the origin address that will transfer its tokens
2. Insert the destination address that will receive the tokens

    !!! note
        The origin and destination addresses must be Ethereum type if the chain is EVM-compatible and Substrate type otherwise.

3. Insert the amount of tokens to transfer
4. Click on **Transfer**

![Transfer Tokens](/images/builders/manage/dapp/token-management/token-management-7.webp)

You'll be asked to sign the transaction with the appchain's governor account. Once the transaction has gone through, the destination account will have received the number of tokens from the origin address.

### Setting Gas Dynamics {: #setting-gas }

> The governor of an EVM-compatible appchain can change its [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md){target=\_blank} configuration, affecting the transaction pricing mechanism. In the **Token Management** panel, click on the **Gas Dynamics** button, and then:

Reviewer: ```suggestion
The governor of an EVM-compatible appchain can change its [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md){target=\_blank} configuration, affecting the transaction pricing mechanism. To do so, in the **Token Management** panel, click on the **Gas Dynamics** button, and then:
```

1. Insert the new base fee, expressed in Wei units (10<sup>-18</sup>)
2. Insert the elasticity value
3. Click on **Update Dynamics**

![Configure Gas Dynamics](/images/builders/manage/dapp/token-management/token-management-8.webp)

You'll be asked to sign the transaction with the appchain's governor account. Once the transaction has gone through, the appchain's fee mechanism will run with the new transaction pricing parameters.

> !!! note

Reviewer 2: Move this note higher up in the ### heading
